activerecord/lib/active_record/relation/where_clause.rb

```diff
@@ -141,13 +139,20 @@ def invert_predicate(node)
         def except_predicates(columns)
           predicates.reject do |node|
             case node
-            when Arel::Nodes::Between, Arel::Nodes::In, Arel::Nodes::NotIn, Arel::Nodes::Equality, Arel::Nodes::NotEqual, Arel::Nodes::LessThan, Arel::Nodes::LessThanOrEqual, Arel::Nodes::GreaterThan, Arel::Nodes::GreaterThanOrEqual
-              subrelation = (node.left.kind_of?(Arel::Attributes::Attribute) ? node.left : node.right)
-              columns.include?(subrelation.name.to_s)
+            when *EXCEPT_PREDICATE_NODES
```

Comment:
This works, but I think it obfuscates the real problem. `Arel.fetch_attribute` doesn't parse attributes from node groups correctly. You can do something like this to fix that more directly with recursion:
```diff
--- a/activerecord/lib/arel.rb
+++ b/activerecord/lib/arel.rb
@@ -46,10 +46,18 @@ def self.arel_node?(value) # :nodoc:
     value.is_a?(Arel::Node) || value.is_a?(Arel::Attribute) || value.is_a?(Arel::Nodes::SqlLiteral)
   end

-  def self.fetch_attribute(value) # :nodoc:
+  def self.fetch_attribute(value, &block) # :nodoc:
     case value
-    when Arel::Nodes::Between, Arel::Nodes::In, Arel::Nodes::NotIn, Arel::Nodes::Equality, Arel::Nodes::NotEqual, Arel::Nodes
::LessThan, Arel::Nodes::LessThanOrEqual, Arel::Nodes::GreaterThan, Arel::Nodes::GreaterThanOrEqual
-      yield value.left.is_a?(Arel::Attributes::Attribute) ? value.left : value.right
+    when Arel::Nodes::Between, Arel::Nodes::In, Arel::Nodes::NotIn, Arel::Nodes::Equality,
+         Arel::Nodes::NotEqual, Arel::Nodes::LessThan, Arel::Nodes::LessThanOrEqual,
+         Arel::Nodes::GreaterThan, Arel::Nodes::GreaterThanOrEqual
+      if value.left.is_a?(Arel::Attributes::Attribute)
+        block.call(value.left)
+      else
+        block.call(value.right)
+      end
+    when Arel::Nodes::Grouping
+      fetch_attribute(value.expr.left, &block) unless value.expr.is_a?(Arel::Nodes::SqlLiteral)
     end
   end
```